torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3006,6 +3025,30 @@ def sample_inputs_entr(op_info, device, dtype, requires_grad, **kwargs):
                                     low=low,
                                     requires_grad=requires_grad)))
 
+
+def sample_inputs_i0_i1(op_info, device, dtype, requires_grad, **kwargs):
```

Comment:
I'm not really sure what's going on here

The comment suggests that i0e is not differentiable at 0, which is OK, but what about expanding the current unary elementwise default to take a tensor generation function and extend make_tensor to take `exclude_zeros` or `exclude=(0,)` so a partial of it can be passed?

Reply:
So right now, AFAIK, `make_tensor` doesn't guarantee that there would be a zero in the output. `exclude` will just guarantee that there wouldn't be zero. So for others `i0`, `i1` `i1e`, we would still need to manually add `zeros` to the tensor. 
So I think it is ok as is for now.

And I think @krshrimali is adding `exclude` functionality in https://github.com/pytorch/pytorch/pull/57941

Reply:
Yep, #57941 adds exclusions to make_tensor(). 

I see, so this wants no zeros if it's i0e, but does want a zero if it's i0, i1, and i1e? 

I guess the more perfect implementation of that pattern would be that we update the unary sample inputs to accept a make_tensor-like function, so i0e could partial make_tensor(..., exclude_zero=True), but that i0, i1, and i1e want to augment the default unary sample inputs with additional samples, including one with zero.

Is that accurate?

Reply:
Yes. Sounds right.

Reply:
OK; would you do me a favor and just add a TODO comment with that thinking above this function? There may be an opportunity to clarify the comment to reflect that thinking, too